### ISSUE TITLE
Cannot add class weights for multilabel classification?

### ISSUE THREAD

**Issue by preethiseshadri518** [keras-team/keras#15984]
I am dealing with a multilabel image dataset where there are 100 labels in total, and each image can have between 1-5 labels. Therefore, the ground truth labels are vectors of length 100, with mostly 0s and 1-5 1s at the label indices. The model has 100 output nodes and uses a sigmoid activation function for each node with binary crossentropy loss.

Since the labels are sparse, I want to up-weight the loss according to each image's label(s). In other words, I want the 1s to be upweighted and 0s to be downweighted by playing around with class weights. However, when I set class_weight={0:0.1, 1:0.9}, get the error:  `IInvalid argument:  indices[0] = 7 is not in [0, 2).`. 

For some reason, I believe it thinks the labels are one-hot encoded multiclass classification labels? Is there a straightforward way to handle this?

```
def create_model(num_classes=100):
    base_model = ResNet50(weights='imagenet', include_top=False)
    x = base_model.output
    x = tf.keras.layers.GlobalAveragePooling2D()(x)
    x = tf.keras.layers.Dropout(0.5, name="dropout")(x)
    dense = tf.keras.layers.Dense(256, activation='relu')(x)
    output = tf.keras.layers.Dense(num_classes, activation='sigmoid', name='output')(dense)
    model = tf.keras.models.Model(inputs=base_model.input, outputs=output)
    return model

model = create_model()
model.fit(
      train_dataset.prefetch(8),
      epochs=epochs,
      class_weight={0.: 0.2, 1.: 0.8},
      steps_per_epoch=len(train_files)//batchsize,
      validation_data=validation_dataset.cache(),
      validation_freq=1,
      callbacks=callbacks,
      verbose=1 
  )
```

**Comment by kumariko** [comment]
@preethiseshadri518 Can you please share a simple standalone code to reproduce the issue? Thanks!
